### PR TITLE
Task/Security related functionality

### PR DESCRIPTION
**Pull request by vikkushnir**
Added security dependencies
Implemented Password Hashing Using Bcrypt
Created JwtUtil, JwtAuthenticationFilter
Implemented JWT authorization
Created AuthenticationController
Implemented login, and registration endpoints

### CODE REVIEW

File: src/main/java/com/ua/accommodation/config/SecurityConfig.java [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+    @Bean
+    public SecurityFilterChain securityFilterChain(HttpSecurity http) throws Exception {
+        return http
+                .cors(AbstractHttpConfigurer::disable)
+                .csrf(AbstractHttpConfigurer::disable)
+                .authorizeHttpRequests(auth -> auth
+                                .requestMatchers("/auth/**",
+                                        "/swagger-ui/**",
+                                        "/v3/api-docs/**")
```
Comment (Nikname2303) [inline src/main/java/com/ua/accommodation/config/SecurityConfig.java:41]: maybe should add /error?

File: src/main/java/com/ua/accommodation/validation/FieldMatch.java [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+@Target({ElementType.TYPE})
+@Retention(RetentionPolicy.RUNTIME)
+@Constraint(validatedBy = FieldMatchValidator.class)
+public @interface FieldMatch {
+    String message() default "Fields must match";
```
Comment (Nikname2303) [inline src/main/java/com/ua/accommodation/validation/FieldMatch.java:14]: can make message "Passwords must be identical"

**Review by Nikname2303 — APPROVED** [review]
grate, some minor comments

**Review by TonyH277 — APPROVED** [review]
Nice!

**Pull request merged**